R/misc.R

```diff
@@ -194,4 +194,43 @@ initial_recipe_call <- function(cl) {
   rec_cl
 }
 
+output_loc <- function(clipboard) {
+  if (clipboard) {
+    res <- tempfile(pattern = "usemodels_")
+  } else {
+    res <- ""
+  }
+  res
+}
 
+route <- function(..., path) {
+  cat(..., "\n\n", file = path, append = path != "")
+  invisible(NULL)
+}
+
+clipboard_output <- function(pth) {
+  if (pth == "") {
+    return(invisible(NULL))
+  }
+  code <- readLines(pth)
+  clipr::write_clip(code, object_type = "character")
+  cli::cli_alert_success("code is on the clipboard.")
+  invisible(NULL)
+}
+
+check_color <- function(cls, clip) {
+  if (cls & clip) {
+    cls <- FALSE
+  }
+  cls
+}
+
+check_clipboard <- function() {
+  # from reprex_clipboard
+  y <- clipr::clipr_available()
+  if (isFALSE(y)) {
+    clipr::dr_clipr()
+    rlang::abort("Please use `clipboard = FALSE`")
```

Comment:
I don't think this logic is right. `clipboard` isn't passed through here, so even if `clipboard = FALSE` it errors if the clipboard isn't available (which I think is what happens in CI)